### PR TITLE
[RHCLOUD-24352] Event type subscription migration

### PR DESCRIPTION
**Pull request by g-duval**

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java [hunk 1]
```diff
@@ -48,9 +50,39 @@ public boolean unsubscribe(String orgId, String username, String bundleName, Str
...
+            .setParameter("userId", username)
+            .setParameter("subscriptionType", subscriptionType.name())
+            .executeUpdate();
+    }
+
+    @Transactional
+    protected int replicateUnsubscribeToEventTypeLevel(String orgId, String username, EmailSubscriptionType subscriptionType, String bundleName, String applicationName) {
+        String query = "DELETE FROM EventTypeEmailSubscription WHERE id.orgId = :orgId AND id.userId = :userId " +
+            "AND id.eventTypeId in (SELECT ev.id FROM EventType ev, Application a, Bundle b WHERE a.bundle.id = b.id and ev.application.id = a.id " +
+            "AND b.name = :bundleName AND a.name = :applicationName) AND id.subscriptionType = :subscriptionType AND id.userId =:userId";
```
[the remark this answers is not in the document]
Comment (g-duval) [reply backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java]: Oups...

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java [hunk 2]
```diff
@@ -48,9 +50,39 @@ public boolean unsubscribe(String orgId, String username, String bundleName, Str
...
+        replicateUnsubscribeToEventTypeLevel(orgId, username, subscriptionType, bundleName, applicationName);
         return true;
     }
 
+    @Transactional
+    protected int replicateSubscribeToEventTypeLevel(String orgId, String username, EmailSubscriptionType subscriptionType) {
+        String query = "INSERT INTO email_subscriptions (user_id, org_id, event_type_id, subscription_type) " +
+            "SELECT ees.user_id, ees.org_id, et.id, ees.subscription_type from endpoint_email_subscriptions ees join event_type et on ees.application_id = et.application_id  " +
+            "WHERE ees.user_id = :userId and ees.org_id = :orgId and ees.subscription_type = :subscriptionType " +
+            "ON CONFLICT (org_id, user_id, event_type_id, subscription_type) DO NOTHING"; // The value is already on the database, this is OK
```
[the remark this answers is not in the document]
Comment (g-duval) [reply backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java]: That's right, I added application and bundle parameters

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java [hunk 3]
```diff
@@ -48,9 +50,44 @@ public boolean unsubscribe(String orgId, String username, String bundleName, Str
...
 
+    @Transactional
+    protected int replicateSubscribeToEventTypeLevel(String orgId, String username, String bundleName, String applicationName, EmailSubscriptionType subscriptionType) {
+        String query = "INSERT INTO email_subscriptions (user_id, org_id, event_type_id, subscription_type) " +
+            "SELECT ees.user_id, ees.org_id, et.id, ees.subscription_type from endpoint_email_subscriptions ees " +
+            "join applications app on ees.application_id = app.id and app.name = :applicationName " +
+            "join bundles bun on app.bundle_id = bun.id and bun.name = :bundleName " +
+            "join event_type et on ees.application_id = et.application_id  " +
+            "WHERE ees.user_id = :userId and ees.org_id = :orgId and ees.subscription_type = :subscriptionType " +
+            "ON CONFLICT (org_id, user_id, event_type_id, subscription_type) DO NOTHING"; // The value is already on the database, this is OK
```
[the remark this answers is not in the document]
Comment (g-duval) [reply backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EmailSubscriptionRepository.java:66]: That's right

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**